src/cdk/a11y/focus-trap/focus-trap.ts

```diff
@@ -194,6 +225,42 @@ export class FocusTrap {
         markers[markers.length - 1] : this._getLastTabbableElement(this._element);
   }
 
+  /**
+   * Focuses the provided element. If the element is not focusable, it will add a tabIndex
+   * attribute to forcefully focus it. The attribute is removed after focus is moved.
+   * @param element The element to focus.
+   */
+  forceFocus(element: HTMLElement, options?: FocusOptions) {
+    if (!this._checker.isFocusable(element)) {
+      element.tabIndex = -1;
+      // The tabindex attribute should be removed to avoid navigating to that element again
+      element.addEventListener('blur', () => element.removeAttribute('tabindex'));
+    }
+    element.focus(options);
+  }
+
+  /**
+   * Focuses the first element that matches the given selector within the focus trap.
+   * @param selector The CSS selector for the element to set focus to.
+   * @returns Whether focus was moved successfully.
+   */
+  focusByCssSelector(selector: string, options?: FocusOptions): boolean {
```

Comment:
If these methods don't live in focus-trap, I don't think you need to return a success/fail boolean